### PR TITLE
[ADD] DIAPAR Improvements:Create products

### PR DESCRIPTION
**Pull request by siddharth7815**
Taks Link:

https://www.druidoo.io/web?db=druidoo-druidoo-master-349765&token=J39sYB1yfc1NMuVYgIBr#id=2169&action=367&model=project.task&view_type=form&menu_id=237

### CODE REVIEW

File: louve_addons/edi_purchase_base/models/supplier_price_list.py [hunk 1]
```diff
@@ -14,9 +14,47 @@ class SupplierPriceList(models.Model):
...
+
+    @api.multi
+    def button_create_product(self):
+        self.ensure_one()
+        # create new product
+        product_tmpl_id = self.env['product.template'].create({'name': self.product_name,
+                                                               'sale_ok': True,
+                                                               'purchase_ok': True,
+                                                               'type': 'product'
+                                                               })
```
Comment (ivantodorovich) [inline louve_addons/edi_purchase_base/models/supplier_price_list.py:32]:
```suggestion
        product_tmpl_id = self.env['product.template'].create({
            'name': self.product_name,
            'sale_ok': True,
            'purchase_ok': True,
            'type': 'product',
        })
```

In the future please use this code-style

File: louve_addons/edi_purchase_base/models/supplier_price_list.py [hunk 2]
```diff
@@ -14,9 +14,47 @@ class SupplierPriceList(models.Model):
...
+        self.env['product.supplierinfo'].create({'name': self.supplier_id.id,
+                                                 'price': self.price,
+                                                 'product_code': self.supplier_code,
+                                                 'product_tmpl_id': product_tmpl_id.id})
+
+        # find similar supplier price list
+        supplier_price_list_ids = self.search([('supplier_id', '=', self.supplier_id.id),
+                                               ('product_name', '=', self.product_name),
+                                               ('supplier_code', '=', self.supplier_code),
+                                               ('product_tmpl_id', '=', False)])
```
Comment (ivantodorovich) [inline louve_addons/edi_purchase_base/models/supplier_price_list.py:46]:
```suggestion
        supplier_price_list_ids = self.search([
            ('supplier_id', '=', self.supplier_id.id),
            ('product_name', '=', self.product_name),
            ('supplier_code', '=', self.supplier_code),
            ('product_tmpl_id', '=', False),
        ])
```

same here.

It results in more readable code and fewer changed lines in future commits

File: louve_addons/edi_purchase_base/models/supplier_price_list.py [hunk 3]
```diff
@@ -14,9 +14,47 @@ class SupplierPriceList(models.Model):
...
+
+        # link product to similar supplier price list
+        supplier_price_list_ids.write({'product_tmpl_id': product_tmpl_id.id})
+
```
Comment (ivantodorovich) [inline louve_addons/edi_purchase_base/models/supplier_price_list.py:50]:
```suggestion
```
Also avoid adding meaningless empty lines.
Comments are great, meaningless empty lines are not

File: louve_addons/edi_purchase_base/models/supplier_price_list.py [hunk 4]
```diff
@@ -14,9 +14,47 @@ class SupplierPriceList(models.Model):
...
+                                                               'type': 'product'
+                                                               })
+        # link product with current supplier price list
+        self.product_tmpl_id = product_tmpl_id.id
+
+        # create product supplier info
+        self.env['product.supplierinfo'].create({'name': self.supplier_id.id,
+                                                 'price': self.price,
+                                                 'product_code': self.supplier_code,
+                                                 'product_tmpl_id': product_tmpl_id.id})
```
Comment (ivantodorovich) [inline louve_addons/edi_purchase_base/models/supplier_price_list.py:40]: Same here, code-style

**Review by ivantodorovich — CHANGES_REQUESTED** [review]
Minor comments on code-style, but looks great
Thanks!

**Comment by ivantodorovich** [comment]
I'm mergin this PR here, code-style is not blocking for 9.0.

Can you forward-port to 12.0 in https://github.com/druidoo/foodcoops ?
(code-style is important there)

**Comment by siddharth7815** [comment]
> I'm mergin this PR here, code-style is not blocking for 9.0.
> 
> Can you forward-port to 12.0 in https://github.com/druidoo/foodcoops ?
> (code-style is important there)

Please have a look Pr for this:  https://github.com/druidoo/FoodCoops/pull/142

**Pull request merged**